cedar-policy-validator/src/validation_result.rs

```diff
@@ -289,15 +327,15 @@ pub struct UnspecifiedEntityError {
 }
 
 /// The structure for validation warnings.
-#[derive(Debug, Clone)]
-pub struct ValidationWarning<'a> {
-    pub(crate) location: SourceLocation<'a>,
+#[derive(Hash, Eq, PartialEq, Error, Debug, Clone)]
```

Comment:
For `ValidationError` this PR hides `Eq` and `PartialEq` behind `cfg(test)`, presumably because the fact that `SourceLocation` factors into the equality is more useful for testing than in other settings.  We should be consistent and either do the same here, or just derive `Eq` and `PartialEq` directly on `ValidationError`.